docs/docs/dev_docs/contracts/portals/data_structures.md

```diff
@@ -0,0 +1,127 @@
+---
+title: Data Structures
+---
+
+The `DataStructures` are structs that we are using throughout the message infrastructure and registry.
+
+**Links**: [Implementation](https://github.com/AztecProtocol/aztec-packages/blob/master/l1-contracts/src/core/libraries/DataStructures.sol).
+
+## `Entry`
+
+An entry for the messageboxes multi-sets. 
+
+```solidity title="DataStructures.sol"
+  struct Entry {
+    uint64 fee;
+    uint32 count;
+    uint32 version;
+    uint32 deadline;
+  }
+```
+
+| Name           | Type    | Description |
+| -------------- | ------- | ----------- |
+| `fee`          | `uint64` | The fee provided to sequencer for including in the inbox. 0 if Outbox (as not applicable). |
+| `count`        | `uint32` | The occurrence of the entry in the dataset |
+| `version`      | `uint32` | The version of the entry |
+| `deadline`     | `uint32` | The consumption deadline of the message. |
+
+
+## `L1Actor`
+
+An entity on L1, specifying the address and the chainid for the entity. Used when specifying sender/recipient with an entity that is on L1.
```

Comment:
```suggestion
An entity on L1, specifying the address and the chainId for the entity. Used when specifying sender/recipient with an entity that is on L1.
```

Reply:
Unsure what exactly we want here, often it is just denoted `chainid` in solidity docs etc, but I prefer `chainId`.